block/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -110,27 +109,26 @@ retrievalWorkerImpl keepAliveTimer SendActions {..} =
         slotDuration <- fromIntegral . toMicroseconds <$> getCurrentEpochSlotDuration
         setTimerDuration keepAliveTimer $ 3 * slotDuration
         startTimer keepAliveTimer
+        -- Exception handlers are installed locally, on the 'thingToDoNext',
+        -- to ensure that network troubles, for instance, do not kill the
+        -- worker.
         thingToDoNext
         mainLoop
```

Comment:
Since we don't have `runIfNotShutdown` in this version, can you reformulate `mainLoop` in terms of `forever`?

Reply:
That's ok as a PR for develop, not needed for a hotfix.